tests/test_flow.py

```diff
@@ -359,6 +360,43 @@ def test_csv_reader_with_key_and_timestamp():
     assert termination_result[1].body == ["m2", datetime(2020, 2, 16, 2, 0), 14, False]
 
 
+@pytest.mark.parametrize(
+    "datetime_field", [{"parse_dates": ["t", "non_existent_column"]}, {"time_field": "non_existent_column"}]
+)
+def test_parse_dates_key_error(datetime_field):
+    with pytest.raises(ValueError) as value_error:
+        controller = build_flow(
+            [
+                CSVSource(
+                    "tests/test-with-timestamp.csv",
+                    header=True,
+                    key_field="t",
+                    **datetime_field,
+                    timestamp_format="%d/%m/%Y %H:%M:%S",
+                )
+            ]
+        ).run()
+        controller.await_termination()
+    assert str(value_error.value) == "Missing column provided to 'parse_dates': 'non_existent_column'"
```

Comment:
Better reformulate the error message, I think:
```suggestion
    assert str(value_error.value) == "Column 'non_existent_column' provided to parse_dates does not exist in the data"
```

Reply:
This is the default pandas.read_csv error message, I did not write it.